packages/fuels-core/src/abi_decoder.rs

```diff
@@ -216,7 +216,7 @@ impl ABIDecoder {
         selected_variant: &ParamType,
         enum_width: usize,
     ) -> Result<Token, CodecError> {
-        // The sway compiler has an optimization where enums that only contain
+        // The FuelVM has an optimization where enums that only contain
```

Comment:
Nit: be consistent about the wording here

Reply:
Do you guys prefer `FuelVM` or `high-level language` ?

Reply:
What bothers me is that, as far as I understand it, the VM could care less about enums. 

It was the sway compiler that had this optimization. But how do you say that without saying sway :D 

High-level-language compiler has an optimization?

Or just be vague, my favourite, "Enums that contain only units are bla bla" don't say why :)

Reply:
That's a good point. In this case, this optimization doesn't happen in the VM. 

I'm ok with what @segfault-magnet said, don't go too deep into why this optimization is done.